src/sortedarray.h

```diff
@@ -0,0 +1,173 @@
+/*
+
+Copyright (c) 2016, Stefan Cloudt
+
+Permission to use, copy, modify, and/or distribute this software
+for any purpose with or without fee is hereby granted, provided
+that the above copyright notice and this permission notice appear
+in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+/**
+ * @file sortedarray.h
+ *
+ * @brief Automatically sorted and resizing array
+ *
+ * SortedArrays are encapsulated ArrayLists which are sorted.
+ *
+ * To create a SortedArray, use @ref sortedarray_new
+ * To destroy a SortedArray, use @ref sortedarray_free
+ *
+ * To add a value to a SortedArray, use @ref sortedarray_prepend, 
+ * @ref sortedarray_append, or @ref sortedarray_insert.
+ *
+ * To remove a value from a SortedArray, use @ref sortedarray_remove
+ * or @ref sortedarray_remove_range.
+ */
+
+#ifndef ALGORITHM_SORTEDARRAY_H
+#define ALGORITHM_SORTEDARRAY_H
+
+#include "arraylist.h"
```

Comment:
A fundamental design decision of this library is that each pair of .c/.h files should be standalone. So it should be possible to take one of these files and just add it into a project without needing to depend on the entire library. Please refactor the code so that this is usable standalone without depending on arraylist.